### DISCUSSION TITLE
Request for instructions for cross-compiling onnxruntime for aarch64

### DISCUSSION THREAD

**Discussion by cravishe** [microsoft/onnxruntime#20430] [General]
Hello,

I'm trying to cross-compile onnxruntime shared object for aarch64 on x86 machine. Kindly help me with the build instructions for it.

**Comment by snnn** [comment]
If you do not use cross-compiling, the instructions are mostly the same as x86. 

See: https://onnxruntime.ai/docs/build/inferencing.html

**Reply by cravishe** [reply]
@snnn , thanks for the reply..
I am planning to cross-compile onnxruntime for aarch64 on x86 machine. What are the steps for cross compilation

**Reply by snnn** [reply]
In the same document as shown above.

**Reply by cravishe** [reply]
@snnn, I used the command `./build.sh --build_shared_lib --config MinSizeRel --arm64 --update --build` for cross-compiling. However, this command generated .so for x86. libonnxruntime.so is generated in _build/Linux/MinSizeRel/_ folder. I used the `file` command in ubuntu to check the architecture and it printed 
`ELF 64 bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked `
 Am I missing anything?

**Comment by jay11ca39** [comment]
@chiru091096 are you able to resolve it..
Even I am facing the same error for aarch64
Used same build command as yours ..
